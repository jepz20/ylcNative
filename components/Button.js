// @flow

import type { StyleObj } from 'react-native/Libraries/StyleSheet/StyleSheetTypes'
import { renderEl } from '../utils'
import * as React from 'react'
import { TouchableHighlight, StyleSheet } from 'react-native'
type Props = {
  children: React.Node,
  style?: StyleObj
}
const Button: React.StatelessFunctionalComponent<Props> = ({
  style,
  children,
  ...props
}) => {
  if (!children) return null

  return (
    <TouchableHighlight
      style={[styles.container, style]}
      underlayColor='rgba(0, 0, 0, 0.4)'
      {...props}
    >
      {renderEl(children)}
    </TouchableHighlight>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  },
  text: {
    color: '#fff',
    fontSize: 40
  }
})

export { Button }
