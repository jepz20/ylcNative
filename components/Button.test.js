import { Text } from 'react-native'
import React from 'react'
import { Button } from './Button'

import renderer from 'react-test-renderer'

describe('Button', () => {
  test('Renders a button with a Text', () => {
    const button = renderer
      .create(
        <Button>
          <Text>Test</Text>
        </Button>
      )
      .toJSON()
    expect(button).toMatchSnapshot()
  })

  test('renders when plain text is provided as a children', () => {
    const button = renderer.create(<Button>Test just string</Button>).toJSON()
    expect(button).toMatchSnapshot()
  })

  test('render nothing no children is provided', () => {
    const button = renderer.create(<Button />).toJSON()
    expect(button).toMatchSnapshot()
  })
})
