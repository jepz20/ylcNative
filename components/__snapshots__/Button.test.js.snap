// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Renders a button with a Text 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={null}
  >
    Test
  </Text>
</View>
`;

exports[`Button render nothing no children is provided 1`] = `null`;

exports[`Button renders when plain text is provided as a children 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={null}
  >
    Test just string
  </Text>
</View>
`;
